editor.planx.uk/src/ui/shared/Input.tsx

```diff
@@ -115,24 +123,32 @@ export default forwardRef((props: Props, ref): FCReturn => {
         container.current?.querySelector("input")?.select();
       },
     }),
-    [],
+    []
   );
 
   return (
     <ErrorWrapper error={errorMessage} id={id}>
-      <StyledInputBase
-        format={format}
-        bordered={bordered}
-        classes={classes}
-        inputProps={{
-          "aria-label": ariaLabel,
-          "aria-describedby": ariaDescribedBy,
-          "aria-labelledby": ariaLabelledBy,
-        }}
-        id={id}
-        ref={container}
-        {...restProps}
-      />
+      <>
+        <StyledInputBase
+          format={format}
+          bordered={bordered}
+          classes={classes}
+          inputProps={{
+            "aria-label": ariaLabel,
+            "aria-describedby": ariaDescribedBy,
+            "aria-labelledby": ariaLabelledBy,
+          }}
+          id={id}
+          ref={container}
+          {...restProps}
+        />
+        {showCharacterCount && (
+          <CharacterCounter
+            count={typeof props.value === "string" ? props.value.length : 0}
+            limit={TEXT_LIMITS[textLength]}
+          />
+        )}
```

Comment:
Yeah, I started off with this strategy but went for a DRY purist view of putting it here to stop repitition. Thinking was that if we ever had a new text field input somewhere, we wouldn't need to think of adding a character count because it is a standard feature of any long text input we use, so embedding the use as standard. 

I do worry about it bloating the <Input> though, so would appreciate the second opinion on it!